src/olympia/addons/models.py

```diff
@@ -1547,14 +1547,16 @@ def block(self):
         from olympia.blocklist.models import Block
 
         # Block.guid is unique so it's either on the list or not.
-        return Block.objects.filter(guid=self.guid).last()
+        guid = getattr(self, 'addonguid', self).guid
```

Comment:
Would it be worth having a helper for this, as it's used twice?

Reply:
mmm, yes.  I did have one at one point along the way, I guess I must have refactored it away.  There's another follow-up refactoring so I'll look to add it there.